### DISCUSSION TITLE
Stuck on firmware 2020.08.04.1602 and in „activation“-loop

### DISCUSSION THREAD

**Discussion by brougs78** [ludwig-v/wireless-carplay-dongle-reverse-engineering#108] [Q&A]
Hi!

I use a CPLAY2air adapter in a KIA Ceed.
I have updated the adpater several month ago and since then after connecting to the car I get a screen to activate the dongle (see screenshot). If I do so (connect to 192.168.50.2 via the mobile phone) the screen changes to show me „Active Success“ and I have to reboot the dongle. After that the Activaton-Screen appears again.

I downgraded to the version 2020.08.04.1602 via the web interface but now I am stuck with this firmware version.
Trying to update via the web interface I only get the message, that the firmware is current.
I also tried via USB stick (tried several versions and naming conventions, several sticks and formating on Mac and Linux) but if I connect the dongle to power (not in the car) I get the solid red LED. If I then attach the USB stick nothing happens … I am not even sure what should happen (flashing red light?).

Is there a way to somehow force the firmware update?
Any ideas.

Thank you very much.

Kind regards,
Andreas

![IMG_4230](https://user-images.githubusercontent.com/90181544/132186014-ee1d7b69-98a6-4bf9-94a4-9569f20a2f9c.jpeg)

**Comment by ludwig-v** [comment]
Hello,

I'm currently working on this issue with my 2 recovered Carlinkit devices.

Try to update to last custom firmware **by USB** first.

--

Carlinkit can disable brands or **even disable your device entirely** because they are the ones sending the content of **/etc/uuid_sign** from their servers which is linked to your device UUID.

That's the part where the activation is broken / disabled for your device, the UUID is returned back instead of the Base64 encoded activation string to put in **/etc/uuid_sign**, that's why you get an activation success message but you are blocked in a forever loop.

**Comment by brougs78** [comment]
Hi Ludwig!

Thanks for the feedback. I tried updating with the latest custom firmware and it worked immediately … very cool.
But unfortunately I am still in the „forever loop“.

Is there something CPLAY2air can do on their end?

Otherwise I will have to wait if you can hack the signing process … do you see a chance?

**Reply by Didi25256** [reply]
Flash this file to get through the activation @brougs78 
U2W_Update_2020.11.10.2138_ROLLBACK.img

**Comment by brougs78** [comment]
Hi Didi!

Thanks for the feedback. This really does break the loop … kind of.
Any Idea what I have to enter here?
![IMG_4539](https://user-images.githubusercontent.com/90181544/138749871-25f9ab13-2bc5-46c9-8095-c34de05ff65f.png)
